src/dmd/expression.d

```diff
@@ -948,7 +948,7 @@ extern (C++) abstract class Expression : RootObject
             }
             else if (!type.isAssignable())
             {
-                error("cannot modify struct `%s` `%s` with `immutable` members", toChars(), type.toChars());
+                error("cannot modify struct instance `%s` of type `%s` because it contains non-mutable members", toChars(), type.toChars());
```

Comment:
How about "value" instead of `struct instance`, or is it important to know it's a struct in the error message?

Reply:
If the struct contains any mutable fields, you can modify the value of the instance by assigning directly to one of the them. Instead, this error message is for instance assignments  (`a = b` for example ) where you modify the value as a whole. In my opinion, this message is better.

Reply:
`where you modify the value as a whole` ...

Reply:
Please use `immutable` not `non-mutable`. The former is the usual term for it, the latter is one I've never seen before.

Reply:
Good point, but then I'd prefer "contains const or immutable members".